src/plugins/lua/force_actions.lua

```diff
@@ -63,6 +64,66 @@ local function gen_cb(expr, act, pool, message, subject, raction, honor, limit,
 
   return function(task)
 
+    local function fill_vars(repl, var_class, var_payload)
+      -- fill template vars prefixed with 
+      --  'selector::' => fill with value extracted by a selector
+      --  'symbol::'   => fill with matching symbol option string(s)
+      --  'task::'     => fill with matching task attribute
+      --                   (only queue_id and uid allowed)
+
+      if var_class == "selector" then
+        local selector = lua_selectors.create_selector_closure(rspamd_config, var_payload, '', true)
```

Comment:
Selectors creation must be done once as this is an expensive procedure that can and will leak memory if used more than one time. So I would suggest to use some module-global cache of selectors like `local sel_cache = {}` and index it when a selector is requested by its name.

Reply:
I see. I will try to use a global cache.